external/dynamo/optimized/processor.py

```
@@ -0,0 +1,818 @@
+                decision_id = data.get("decision_id")
+                break
+
+            # Record routing decision
+            if worker_id is not None:
+                self._metrics.routing_decisions_total.inc({"worker_id": str(worker_id)})
+            else:
+                logger.warning("Router stream ended without worker_id; falling back to engine load balancing.")
+
+            return worker_id, decision_id
+
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)